### PR TITLE
Added many more root specification properties

### PR DESCRIPTION
**Pull request by mattbischoff**
The only two left are “license” and “source”.

I also added the necessary tests for these and introduces `TMAuthor` to model a Podspec author / maintainer. Super open to feedback on this.

As to why I did it, I just wanted to do some braindead typing this morning. No idea why.

### CODE REVIEW

File: Pod/Classes/TMPodspec.m [hunk 1]
```diff
@@ -61,21 +76,121 @@ - (NSString *)version {
...
+                [screenshotURLs addObject:URL];
+            }
+        }
+    }
+    else if (singleScreenshotURLString) {
+        NSURL *URL = [NSURL URLWithString:singleScreenshotURLString];
+        [screenshotURLs addObject:URL];
+    }
+    
+    return [screenshotURLs count] ? screenshotURLs : nil;
```
Comment (segiddins) [inline Pod/Classes/TMPodspec.m]: why not return the empty array?

Comment (mattbischoff) [inline Pod/Classes/TMPodspec.m]: Because I want to represent that the spec author did not include screenshots.

File: Pod/Classes/TMPodspec.m [hunk 2]
```diff
@@ -61,21 +76,125 @@ - (NSString *)version {
...
+                TMAuthor *author = [[TMAuthor alloc] initWithName:(NSString *)name emailAddress:(NSString *)emailAddress];
+                [authors addObject:author];
+            }
+        }];
+    }
+    else if (authorNames) {
+        for (id name in authorNames) {
+            if ([name isKindOfClass:[NSString class]]) {
+                TMAuthor *author = [[TMAuthor alloc] initWithName:(NSString *)name emailAddress:nil];
+                [authors addObject:(NSString *)author];
```
Comment (irace) [inline Pod/Classes/TMPodspec.m]: This cast is wrong, right?

File: Pod/Classes/TMPodspec.m [hunk 3]
```diff
@@ -61,21 +76,125 @@ - (NSString *)version {
...
+            }
+        }
+    }
+    else if (authorName) {
+        TMAuthor *author = [[TMAuthor alloc] initWithName:authorName emailAddress:nil];
+        [authors addObject:(NSString *)author];
```
Comment (irace) [inline Pod/Classes/TMPodspec.m]: Same here.

File: Pod/Classes/TMPodspec.m [hunk 4]
```diff
@@ -61,21 +76,125 @@ - (NSString *)version {
...
+    }
+    else {
+        return nil;
```
Comment (irace) [inline Pod/Classes/TMPodspec.m]: Don't feel *too* strongly but why not just return the empty array in this case? Similar to what `NSFetchRequest` does.

File: Pod/Classes/TMPodspec.m [hunk 5]
```diff
@@ -61,21 +76,125 @@ - (NSString *)version {
...
+        }
+    }
+    else if (singleScreenshotURLString) {
+        NSURL *URL = [NSURL URLWithString:singleScreenshotURLString];
+        if (URL) {
+            [screenshotURLs addObject:URL];
+        }
+    }
+    
+    return [screenshotURLs count] ? screenshotURLs : nil;
```
Comment (irace) [inline Pod/Classes/TMPodspec.m]: Same here, should we return the empty array?

File: Pod/Classes/TMPodspec.m [hunk 6]
```diff
@@ -61,21 +76,125 @@ - (NSString *)version {
     return [self stringForKey:PodspecKeyVersion];
 }
 
-- (NSString *)summary {
-    return [self stringForKey:PodspecKeySummary];
+- (NSArray *)authors {
+    NSString *authorName = [self stringForKey:PodspecKeyAuthor];
+    NSArray *authorNames = [self arrayForKey:PodspecKeyAuthors];
+    NSDictionary *authorNamesAndAddresses = [self dictionaryForKey:PodspecKeyAuthors];
```
Comment (irace) [inline Pod/Classes/TMPodspec.m]: I don't feel strongly but wondering if it's not worth creating the `authorNames` and `authorName` local variables until it's been deduced that `authorNamesAndAddresses` doesn't exist... ¯\_(ツ)_/¯

File: Pod/Classes/TMPodspec.m (hunk 4 above)
Comment (mattbischoff) [inline Pod/Classes/TMPodspec.m]: Because I want to give the caller some idea as to whether the author included the key screenshots or not. If they get an empty array, it’s because the author put the wrong kind of data in the array, if they get nil, it means it wasn’t included at all.

File: Pod/Classes/TMPodspec.m (hunk 2 above)
Comment (mattbischoff) [inline Pod/Classes/TMPodspec.m]: Whoops, fixed both. Copy paste got me.

File: Pod/Classes/TMPodspec.h [hunk 7]
```diff
@@ -18,9 +18,19 @@
 @property (nonatomic, readonly) NSString *version;
 
 /**
- *  @see http://guides.cocoapods.org/syntax/podspec.html#summary
+ *  @see http://guides.cocoapods.org/syntax/podspec.html#authors
  */
-@property (nonatomic, readonly) NSString *summary;
+@property (nonatomic, readonly) NSArray /* <TMAuthor> */ *authors;
```
Comment (irace) [inline Pod/Classes/TMPodspec.h]: About as pedantic as can be, but I'd prefer if the spaces were removed to make it ``NSArray/*<TMAuthor>*/ *authors;`, just because it'd be slightly closer to how it'd read in a language like Swift or Java with actual generics (`Array<Author>`)

Comment (mattbischoff) [inline Pod/Classes/TMPodspec.h]: I based the spacing on UIAlertView.h’s which is for a different purpose I guess, so I can change this.

```objc
@property(nonatomic,assign) id /*<UIAlertViewDelegate>*/ delegate;    // weak reference
```

Comment (mattbischoff) [inline Pod/Classes/TMPodspec.h]: Sure.

File: Pod/Classes/TMPodspec.m (hunk 6 above)
Comment (mattbischoff) [inline Pod/Classes/TMPodspec.m]: That’d increase the nesting if I do it the way I’m thinking about, but yeah, we could. Which part of them being here bugs you: performance concerns, too many things in scope, something else?

File: Pod/Classes/TMPodspec.m (hunk 5 above)
Comment (mattbischoff) [inline Pod/Classes/TMPodspec.m]: I want to represent what the author put in their podsec:

1. If you get back `nil` it means they didn’t include screenshots.
2. If you get back an empty array, it means they tried to but either included them incorrectly or we parsed incorrectly (likely due to some change in the spec).

At least that’s how I thought about it.

File: Example/Tests/Tests.m [hunk 8]
```diff
@@ -23,14 +25,51 @@ - (void)testSpecVersion {
     XCTAssertEqualObjects(self.spec.version, @"1.0.0");
 }
 
-- (void)testSpecSummary {
-    XCTAssertEqualObjects(self.spec.summary, @"A library with some functionality");
+- (void)testSpecAuthors {
+    XCTAssertEqualObjects(self.spec.authors, @[[[TMAuthor alloc] initWithName:@"Bryan Irace" emailAddress:@"bryan@irace.me"]]);
```
Comment (irace) [inline Example/Tests/Tests.m]: Probably worth adding a couple of cases for the other ways of specifying authors.

File: Pod/Classes/TMPodspec.m (hunk 4 above)
Comment (irace) [inline Pod/Classes/TMPodspec.m]: OK, works for me.

File: Pod/Classes/TMPodspec.m (hunk 6 above)
Comment (irace) [inline Pod/Classes/TMPodspec.m]: Yeah just extra stuff in scope. It's fine though, more just me thinking out loud than anything else.

File: Example/Tests/Tests.m (hunk 8 above)
Comment (mattbischoff) [inline Example/Tests/Tests.m]: Agreed. How would you do that, just manufacture dictionaries or include multiple sample JSON files?

File: Pod/Classes/TMPodspec.h [hunk 9]
```diff
@@ -28,6 +38,41 @@
...
+
+/**
+ *  @see http://guides.cocoapods.org/syntax/podspec.html#description
+ */
+@property (nonatomic, readonly) NSString *descriptionString;
+
+/**
+ *  @see http://guides.cocoapods.org/syntax/podspec.html#screenshots
+ */
+@property (nonatomic, readonly) NSArray /* <NSURL> */ *screenshotURLs;
```
Comment (irace) [inline Pod/Classes/TMPodspec.h]: One more: `NSArray/*<NSURL>*/`

File: Example/Tests/Tests.m (hunk 8 above)
Comment (irace) [inline Example/Tests/Tests.m]: Personally I think having a single JSON file that has at least one variant for each field, plus manufactured partials for verifying alternate behavior is a fine balance. What do you think?

Comment (mattbischoff) [inline Example/Tests/Tests.m]: Took a crack at this in 688bfae. Let me know what you think

File: Pod/Classes/TMPodspec.m (hunk 6 above)
Comment (irace) [inline Pod/Classes/TMPodspec.m]: Not necessary to do now but might be worth doing stuff like these in individual classes, like have a class that takes `self.dictionary` and returns an authors array, another that takes `self.dictionary` and returns a screenshots array, etc.

Comment (mattbischoff) [inline Pod/Classes/TMPodspec.m]: Yeah, agreed. Started out simple in this class and grew more complex. Make an issue for it and I’ll do it in a separate PR?

Comment (irace) [inline Pod/Classes/TMPodspec.m]: :+1:
